meta/client/tidbclient/object.go

```diff
@@ -296,6 +296,121 @@ func (t *TidbClient) ReplaceObjectMetas(object *Object, tx Tx) (err error) {
 	return
 }
 
+func (t *TidbClient) AppendObject(object *Object, updateUsage bool) (err error) {
+	tx, err := t.Client.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+		}
+		if err != nil {
+			tx.Rollback()
+		}
+	}()
+
+	sql, args := object.GetCreateSql()
+	_, err = tx.Exec(sql, args...)
+
+	sql, args = object.GetCreateHotSql()
+	_, err = tx.Exec(sql, args...)
+
+	if updateUsage {
+		err = t.UpdateUsage(object.BucketName, object.Size, tx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *TidbClient) ScanHotObjects(limit int, bMarker, oMarker, vMarker string) (result ScanHotObjectsResult, err error) {
+	var count int
+	var customattributes, acl, lastModifiedTime string
+	var sqltext string
+	var rows *sql.Rows
+	if bMarker == "" && oMarker == "" && vMarker == "" {
+		sqltext = "select bucketname,name,version,location,pool,ownerid,size,objectid,lastmodifiedtime,etag,contenttype," +
+			"customattributes,acl,nullversion,deletemarker,ssetype,encryptionkey,initializationvector,type,storageclass,createtime" +
+			" from hotobjects order by bucketname,name,version limit ?;"
+		helper.Logger.Info("sqltext0", sqltext)
+		rows, err = t.Client.Query(sqltext, limit)
+	} else {
+		sqltext = "select bucketname,name,version,location,pool,ownerid,size,objectid,lastmodifiedtime,etag,contenttype," +
+			"customattributes,acl,nullversion,deletemarker,ssetype,encryptionkey,initializationvector,type,storageclass,createtime" +
+			" from hotobjects where bucketname>=? and name >=? and version >=? order by bucketname,name,version limit ?;"
```

Comment:
```
A_bucket_A_object_A_version
A_bucket_A_object_B_version <- cursor
A_bucket_B_object_A_version
```
Suppose cursor is at `A_bucket_A_object_B_version`, `A_bucket_B_object_A_version` would be skipped.

Reply:
In the previous design, the first row should be `A_bucket_A_object_B_version` after next query, and this row would be skiped.